File: lee/src/main/java/org/example/lee/题目/链表/a_反转链表.go
package 链表

func reverseList(head *ListNode) *ListNode {
	n := head
	var h, h1 *ListNode
	for n != nil {
		h = n
		n = n.Next
		h.Next = h1
		h1 = h
	}
	return h1
}

File: lee/src/main/java/org/example/lee/题目/链表/a_反转链表.java
package org.example.lee.题目.链表;

import org.example.lee.model.ListNode;

public class a_反转链表 {
	public ListNode reverseList(ListNode head) {
		ListNode h1 = null, h2 = null, h3 = null;
		while (head != null) {
			h1 = head.next;
			head.next = h2;
			h2 = head;
			head = h1;
		}
		return h2;
	}
}
